ambry-rest/src/main/java/com.github.ambry.rest/NettyResponseChannel.java

```diff
@@ -60,6 +62,10 @@
  * will be accepted and all scheduled writes will be notified of the failure.
  */
 class NettyResponseChannel implements RestResponseChannel {
+  // add to this list if the connection needs to be closed on certain errors on GET, DELETE and HEAD.
+  // for a POST, we always close the connection on error because we expect the channel to be in a bad state.
+  static final List<HttpResponseStatus> CLOSE_CONNECTION_ERROR_STATUSES = new ArrayList<>();
```

Comment:
Ah ok. No it won't. AFAIK clients close connections in HTTP 1.1 when
1. Remote peer resets the connection (i.e. closes it).
2. Client experiences a connection error (not an error status code - an actual error with the connection).
3. If the server sends the header "Connection:close" in the response.

Otherwise, connections are kept open by default.

Also the current frontend we have already does all of this - only differences are 
1) We close for POST no matter what the error is (current closes only if BAD_REQUEST because it pulls the whole request into memory before working on it. Therefore at the time of the error, the connection is "clean").
2) We don't close for BAD_REQUEST (I believe the current one does this because this is the only error that is thrown before POST pulls the whole data in).